### DISCUSSION TITLE
[2024-02-12] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#9031] [Weekly Demo Day]
Quite a short week, did some fixes:
* #8994 - i.e. the boolean does not drop-down attached (unless it have more options than True/False) and it inserts `True`/`False` intead of `Boolean.True`/`Boolean.False`.
* #8990 - fixes the nodes from old project being visible for a moment.

[Screencast from 2024-02-07 14-53-00.webm](https://github.com/enso-org/enso/assets/3919101/2a58bf11-05ef-4049-9d4d-2cd2d3ed469f)

* #8978 - the "Geospatial Analysis" should not be broken anymore.

**Comment by MichaelMauderer** [comment]
* Finished work on e2e tests #8989 
* Working on full wheel component menu #9029

[Peek 2024-02-12 14-11.webm](https://github.com/enso-org/enso/assets/1428930/80febd9e-87ab-4843-b09c-e1aab826b71a)

**Comment by vitvakatu** [comment]
Presented last demo:
- https://github.com/enso-org/enso/pull/8950
- https://github.com/enso-org/enso/pull/8983

Working on 
- https://github.com/enso-org/enso/issues/5353, should be available by Tuesday/Wednesday.

**Comment by kazcw** [comment]
Working on #8238:
- Implemented replacement for GUI1's apply-text-edits algorithm.
- Fixed problems with our CodeMirror integration (Fixed view stability; Fixed a focus bug; Fixed errors caused by diagnostics range exceptions).
Fixing one last CM problem now.

**Comment by Frizi** [comment]
- Finished the figma icons saga, shared the export plugin #8387
	- we may want to create a private github repo for it, as for now it is shared by tossing a zip around
- Currently debugging a performance regression while editing the nodes on a project containing about 10 nodes

![image](https://github.com/enso-org/enso/assets/919491/0d99b4c3-f464-431b-8efb-a5139d4132fb)

**Comment by mwu-tow** [comment]
* Last week fixes seem to have improved the situation (#8970, #8967, #8947, #8918)
* Still more fixes are needed — please make sure that any issues are reported.
* Tweaks for Engine tests #9001.
* Spend much time debugging code signing issues on macOS.
  * For now we'll disable one of the runners.
  * After removing the old GUI it might be worth to reconsider.

**Comment by JaroslavTulach** [comment]
Engine progress:

- Let project manager work against any folder #8985
- [Official results](https://enso-org.github.io/engine-benchmark-results/stdlib-benchs.html) of benchmarking integration of #8924 indicate speedup by more than 100ms!

![obrazek](https://github.com/enso-org/enso/assets/26887752/7f2dd572-0ab3-4ceb-897c-e1be8d9543fe)

- E.g. the speedup isn't just because of #8698
- @ola-lis updated [Enso VSCode extension](https://marketplace.visualstudio.com/items?itemName=Enso.enso4vscode) to version 1.33.29 that provides support for **debugging a single Enso file**:

![Select enso executable](https://github.com/enso-org/enso/assets/26887752/4e1d0666-634d-4fb8-bf61-6dbf765311e8)

- Read more about our VSCode tooling in the [documentation](https://github.com/enso-org/enso/blob/d87ba6e3cf7804e72909db1a984e1b53c4b0c004/tools/enso4igv/README.md#debugging-a-single-enso-file)!
